space-framework/docs/space-framework.md

```diff
@@ -0,0 +1,92 @@
+
+The space framework uses two CRDs:
+- **SpaceProviderDesc:** This CRD represents a space provider it holds the information needed in order to interact with a specific Space  provider. 
+- **Space:** This CRD represents a space as defined above. It includes reference to secrets that allow clients to connect to the pSpace that it represents and also several attributes used by the SF that will be described later on. 
+
+Spaces can be divided into two categories:
+1. Provider based spaces: The Space Manager (SM) uses a space provider to get information (or perform operations) on the spaces. We can think of such pSpaces as "belong" to the space provider. There are two types of spaces in this category which control the way these spaces are added to the space framework: 
+   * **Managed Space**: The pSpace is created through an explicit request to the management api server to create a new Space. For Managed spaces the space framework initiates a request to the Space provider to create the pSpace, the space frame work will then continuously reconcile between the Space and the pSpace states. For managed spaces the desired state is controlled through the Space object. For example, in order to delete the pSpace the client need to delete the corresponding Space object and the SF will interact with the space provider in order to delete the pSpace.
+   * **Unmanaged Space**: The SF supports discovery of existing pSpaces on a space provider and adding them as "Unmanaged Spaces" into the framework. This mode is similar to the import mode in terms of the "source of truth" - the desired state is defined by the pSpace. However, as oppose to import, here the pSpaces are automatically discovered through the space provider regular interface and the pSpace access information is retrieved from the  space provider similarly to how it is done in the managed space case. The SF supports pattern based filtering rules to limit the discovery of pSpaces. 
+  2. Providerless spaces: This spaces don't belong to a space provider and are called "Imported spaces". Such Space is created by importing an existing pSpace. In this case the Space is not linked to any space provider and the client which imports that Space need to supply the access information that allows other clients to connect to the pSpace. For imported spaces the desired state is derived by the pSpace. This means that the SF is not responsible for the life cycle management of the pSpace but only update the corresponding Space object. For example, when deleting the space object of an existing imported pSpace the space framework will recreate the space object. 
```

Comment:
> This means that the SF is not responsible for the life cycle management of the pSpace but only update the corresponding Space object.

Update based on what? What updates are we talking about here?

> For example, when deleting the space object of an existing imported pSpace the space framework will recreate the space object.

When _who_ is deleting the Space object? I think you are trying to say that if anyone deletes the Space object then it will be re-created with the same content as it originally had. Is that right? I still find this surprising. Why is all that information being stored redundantly? Is there a way for the user (or an authorized colleague) who originally imported a space to undo that?

Reply:
I am not sure I understand that answer. If a user deletes a Space object, does that undo the import? Your earlier text seems to say "no, the SM will re-create the Space object". Which is it? Are there other cases where the SM updates the Space object? If the SM will re-create an imported Space object after it is deleted, why was the SM keeping all the needed information somewhere else?

Reply:
OK, let me clarify. There is a difference between Import (providerless) and unmanaged spaces (using discovery over a provider):
For Imported spaces - deletion of the space object is basically an "undo import" the Space object will not be re-created
For unmanaged spaces - if the Space is deleted, the SM will recreate it if the pSpace exists. 
I think I indeed made a mistake in the description of Imported spaces. Will fix

Reply:
Yes, please do.

Reply:
Already did, last week. Hopefully it's OK

Reply:
In https://github.com/ezrasilvera/edge-mc/blob/bccd082b5f4ad29740a668e8bb987bc0d18bde4e/space-framework/docs/space-framework.md?plain=1 I still see "For example, when deleting the space object of an existing imported pSpace the space framework will recreate the space object".